apps/api/tasks.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from uuid import UUID
+
+from celery.app import shared_task
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Subquery
+from taskbadger.celery import Task as TaskbadgerTask
+
+from apps.channels.clients.connect_client import CommCareConnectClient
+from apps.channels.models import ChannelPlatform, ExperimentChannel
+from apps.experiments.models import Experiment, ParticipantData
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(bind=True, base=TaskbadgerTask, ignore_result=True)
+def setup_connect_channels_for_bots(self, connect_id: UUID, experiment_data_map: dict):
+    """
+    Set up Connect channels for experiments that are using the ConnectMessaging channel
+
+    experiment_data_map: {experiment_id: participant_data_id}
+    """
+
+    experiment_ids = list(experiment_data_map.keys())
+    participant_data_ids = list(experiment_data_map.values())
```

Comment:
since we have these IDs, do we need to add the experiment filter below?

Reply:
[This](https://github.com/dimagi/open-chat-studio/blob/c6574b4b53844b32da0f73689b18aee582d0580d/apps/api/tasks.py#L31-L34) filter? This is filtering the that list of experiments for those that has a `commcare_connect` channel linked. Ideally the resulting experiment id list should be equal to the incoming list, but in the case where it's not, we probably don't want to setup channels that would not be used.